### DISCUSSION TITLE
Error while compiling in Arduino

### DISCUSSION THREAD

**Discussion by login9903** [luc-github/ESP3D#615] [Questions]
Hi,

i get an error when compiling in Arduino : sketch\src\modules\websocket\websocket_server.cpp:28:30: fatal error: WebSocketsServer.h: No such file or directory
 #include <WebSocketsServer.h>

The WebSocketsServer.h is missing from the directory src/modules/websocket

I also tried with Platformio but i get a bunch of errors that i cannot understand

Compiling for an MKS Robin Nano V3, so ESP8266 board

How can i solve this problem ?

Thank you!

**Comment by luc-github** [comment]
v3 installation process is same as v2.1 https://github.com/luc-github/ESP3D#installation
use libraries present in repository

**Reply by login9903** [reply]
I need to copy the library folder from the V2.1 or V3 repository?

**Comment by luc-github** [comment]
each version / branch  have libraries directory in repository so use the one from the version you are using
but instalation steps are identicals between 2.1 and 3.0
